File: JMP.Enterprises/frontend/src/types/expense.ts
export interface Expense {
  expenseId: number;
  propertyId?: number;
  propertyName?: string;
  expenseDate: string;
  category: 'Maintenance' | 'Utilities' | 'Cleaning' | 'HOA Fees' | 'Taxes' | 'Supplies' | 'Other' | string;
  amount: number;
  vendorPayee?: string;
  description: string;
  receiptReference?: string;
  createdDate: string;
}

export interface CreateExpenseDto {
  propertyId?: number;
  expenseDate: string;
  category: string;
  amount: number;
  vendorPayee?: string;
  description: string;
  receiptReference?: string;
}

export interface UpdateExpenseDto {
  propertyId?: number;
  expenseDate: string;
  category: string;
  amount: number;
  vendorPayee?: string;
  description: string;
  receiptReference?: string;
}

export interface ExpenseCategorySummary {
  category: string;
  totalAmount: number;
  expenseCount: number;
  percentage: number;
}

File: JMP.Enterprises/backend/JMP.Enterprises.Api/Models/Expense.cs
using System;

namespace JMP.Enterprises.Api.Models;

/// <summary>
/// Represents an expense / operational cost incurred for a property or general operation.
/// Maps to "Expenses" table in SQL Server.
/// </summary>
public class Expense
{
    public int ExpenseId { get; set; }

    public int? PropertyId { get; set; }
    public Property? Property { get; set; }

    public DateTime ExpenseDate { get; set; } = DateTime.Now;

    /// <summary>
    /// Maintenance | Utilities | Cleaning | HOA Fees | Taxes | Supplies | Other
    /// </summary>
    public string Category { get; set; } = "Maintenance";

    public decimal Amount { get; set; }

    public string? VendorPayee { get; set; }

    public string Description { get; set; } = string.Empty;

    public string? ReceiptReference { get; set; }

    public DateTime CreatedDate { get; set; } = DateTime.Now;
}
